### PR TITLE
chore: Initial structure and implementation of FDv2 datasystem

### PR DESCRIPTION
**Pull request by keelerm84**
There is still a lot of work to be done to get this functional, but this gives us some shape to the remaining tasks.

---

> [!NOTE]
> Introduce experimental FDv2 data system (store, polling/streaming, status) with new DataSystemConfig and integrate selection into client; update protocols/APIs and tests.
> 
> - **Client/Core**:
>   - Integrates new `FDv2` data system; selects `FDv1` when `config.datasystem_config` is absent, otherwise `FDv2` (honors `offline`).
>   - Initialization readiness now based on `data_availability.at_least(DataAvailability.CACHED)`.
>   - Wires data system providers (`data_store_status_provider`, `data_source_status_provider`, `flag_tracker`, `store`) and passes diagnostic accumulator.
> - **Datasystem v2**:
>   - Adds `ldclient.impl.datasystem.fdv2.FDv2` with initializer/synchronizer orchestration, fallback/recovery logic, availability, and threading.
>   - New `Store` for FDv2 handling apply/commit, selector, dependency tracking, and flag change notifications.
>   - New `datasourcev2.status.DataSourceStatusProviderImpl`.
>   - `datasourcev2.polling` and `datasourcev2.streaming`: expose `name` property; implement sync/fetch behaviors and error handling.
>   - Protocol updates: `Initializer` and `Synchronizer` now require `name` property; add to `StreamingDataSource` and test data source.
> - **Configuration**:
>   - Public `DataSystemConfig` dataclass added to `ldclient.config` plus `datasystem_config` field on `Config`.
>   - `impl/datasystem/config.py` builder now produces `DataSystemConfig`; supports default/streaming/polling/custom setups.
> - **FDv1**:
>   - `data_availability` computed dynamically based on processor/store state.
> - **Tests**:
>   - Add tests for FDv2 orchestration and availability; update polling initializer/synchronizer tests (use `name` property, expanded error cases);
>   - Add config builder tests including immutability and None/empty initializer cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a5d7c728f121b1d6a24a140bbcc81565513b6ef1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: ldclient/client.py [hunk 1]
```diff
@@ -249,14 +251,19 @@ def __start_up(self, start_wait: float):
...
+            )
+
+            self._data_system: DataSystem = FDv1(self._config)
+        else:
+            self._data_system = FDv2(datasystem_config, disabled=self._config.offline)
 
-        self._data_system = FDv1(self._config)
         # Provide flag evaluation function for value-change tracking
-        self._data_system.set_flag_value_eval_fn(
+        self._data_system.set_flag_value_eval_fn(  # type: ignore
```
Comment (keelerm84) [inline ldclient/client.py]: These ignores are because the data system protocol doesn't expose these methods, but both fdv1 and fdv2 implementations do. So we have to tell the linter to trust us in a few places.

File: ldclient/impl/datasystem/fdv2.py [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+        self._threads.append(main_thread)
+
+    def stop(self):
+        """Stop the FDv2 data system and all associated threads."""
+        self._stop_event.set()
+
+        # Wait for all threads to complete
+        for thread in self._threads:
+            if thread.is_alive():
+                thread.join(timeout=5.0)  # 5 second timeout
```
Comment (kinyoklion) [inline ldclient/impl/datasystem/fdv2.py]: Envious of join with timeout. Do we need to log anything in the timeout case? (Though the is_alive thing really feels questionable, so maybe not.)
